smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/TypeScriptWriter.java

```diff
@@ -153,6 +153,17 @@ public TypeScriptWriter addImport(String name, String as, PackageContainer from)
         return this.addImport(name, as, from.getPackageName());
     }
 
+    /**
+     * Same as {@link #addImport(String, String, PackageContainer)} but appends a
+     * submodule path, for example "@smithy/core/cbor".
+     */
+    public TypeScriptWriter addImportSubmodule(String name, String as, PackageContainer from, String submodule) {
+        if (from instanceof Dependency) {
```

Comment:
Suggestion (also just excited about any java features past 8): You can use instanceof pattern matching here: https://blogs.oracle.com/javamagazine/post/pattern-matching-for-instanceof-in-java-14